src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1457,6 +1457,10 @@ private IForEachLoopOperation CreateBoundForEachStatementOperation(BoundForEachS
                 var declaratorSyntax = boundForEachStatement.IterationVariableType.Syntax;
                 loopControlVariable = new Lazy<IOperation>(() => new VariableDeclarator(local, initializer: null, ignoredArguments: ImmutableArray<IOperation>.Empty, semanticModel: _semanticModel, syntax: declaratorSyntax, type: null, constantValue: default, isImplicit: false));
             }
+            else if (boundForEachStatement.IterationErrorExpressionOpt != null)
```

Comment:
>if (boundForEachStatement.IterationErrorExpressionOpt != null) [](start = 17, length = 62)

I think we should make more changes to this ```if... else if ... else``` statement. I we have DeconstructionOpt, then loopControlVariable is DeconstructionOpt. If we have IterationErrorExpressionOpt, then loopControlVariable  is IterationErrorExpressionOpt. Only otherwise we should synthesize node for loopControlVariable and I am not even sure if today's final ```else``` should even be reachable.